subsys/pm/policy.c

```diff
@@ -90,6 +91,11 @@ const struct pm_state_info *pm_policy_next_state(uint8_t cpu, int32_t ticks)
 		const struct pm_state_info *state = &cpu_states[i];
 		uint32_t min_residency, exit_latency;
 
+		/* skip a state that set min_residency to FOREVER .*/
+		if (state->min_residency_us == FOREVER) {
```

Comment:
We already have `SYS_FOREVER_US`, why not re-use it? Btw, I don't think this should be the _standard_ mechanism to skip certain states, there should be an API to do that.

Reply:
I was just bringing attention to this, the constant itself can be anything, `FOREVER` is a terrible name anyways. Regarding API, we do have it through constraints, but the problem is how do you represent such state in DT ? It has to be a way to declare that. I see some options:

1) states without min_residency_us property are excluded by the policy and can only be used though the force API
	- That is an implicit behavior and may not be clear, though without `min_residency` properly set, that is probably the right thing to do.
2) A new property to state that the state is disabled (the problem is that is not technically disabled, just that the policy will not account it)
	- This required an additional field in the struct
3) Setting the `min_residency_us` to a constant value like any of those `forever` that we have.

I am kind in favor of option 1 now.

Reply:
@gmarull @JordanYates ^